src/CP/core/model.jl

```diff
@@ -257,6 +257,8 @@ Empty the CPModel.
 """
 function Base.empty!(model::CPModel)
     empty!(model.variables)
+    empty!(model.branchable_variables)
+    empty!(model.branchable)

```

Comment:
These lines fix a major bug caused by the fact that the model was not properly emptied after the resolution of each instance in the training process.

Reply:
Nice spot !